azure/Guidelines.md

```diff
@@ -0,0 +1,864 @@
+
+> :white_check_mark: **DO** sort the collection in ascending order on an expression if "asc" or "desc" is not specified.
+> 
+> :white_check_mark: **DO** sort NULL values as "less than" non-NULL values.
+> 
+> :white_check_mark: **DO** sort items by the result values of the first expression, and then sort items with the same value for the first expression by the result value of the second expression, and so on.
+> 
+> :white_check_mark: **DO** use the inherent sort order for the type of the field. For example, date-time values should be sorted chronologically and not alphabetically.
+> 
+> :white_check_mark: **DO** respond with an error message as defined in the [Handling Errors](#Handling-errors) section if the client requests sorting by a field that is not supported by the operation.
+
+For example, to return all people sorted by name in ascending order:
+```http
+GET https://api.contoso.com/people?orderby=name
+```
+
+For example, to return all people sorted by name in descending order and a secondary sort order of hireDate in ascending order.
+```http
+GET https://api.contoso.com/people?orderby=name desc,hireDate
+```
+
+Sorting MUST compose with filtering such that:
+```http
+GET https://api.contoso.com/people?filter=name eq 'david'&orderby=hireDate
+```
+will return all people whose name is David sorted in ascending order by hireDate.
+
+##### Considerations for sorting with pagination
+
+> :white_check_mark: **DO** use the same filtering options and sort order for all pages of a paginated list operation response.
+
+##### skip
+> :white_check_mark: **DO** define the _skip_ parameter as an integer with a default and minimum value of 0.
+
+> :heavy_check_mark: **YOU MAY** allow clients to pass the _skip_ query parameter to specify an offset into collection of the first resource to be returned.
+##### top
+
+> :heavy_check_mark: **YOU MAY** allow clients to pass the _top_ query parameter to specify the maximum number of resources to return from the collection.
+
+If supporting _top_:
+> :white_check_mark: **DO** define the _top_ parameter as an integer with a minimum value of 1. If not specified, __top__ has a default value of infinity.
+> 
+> :white_check_mark: **DO** return the collection's _top_ number of resources (if available), starting from _skip_.
+
+##### maxpagesize
+
+> :heavy_check_mark: **YOU MAY** allow clients to pass the _maxpagesize_ query parameter to specify the maximum number of resources to include in a single page response.
+
+If supporting _maxpagesize_
+
+> :white_check_mark: **DO** define the _maxpagesize_ parameter as an optional integer with a default value appropriate for the collection.
+> 
+> :white_check_mark: **DO** make clear in documentation of the _maxpagesize_ parameter that the operation may choose to return fewer resources than the value specified.
+
+### API Versioning
```

Comment:
OSS
- Semantic versioning is popular in OSS, do we have an opinion on that pattern in this doc? 
- Some Microsoft services proxy calls to OSS Rest APIs. These will have their own semantics. Are there any considerations for how to frame/expose these within an API following MS conventions?

Reply:
On semantic versioning, we're requiring services to always be backwards compatible which obviates a lot of the need for semver.  We prefer date based versioning as outlined below and in https://www.youtube.com/watch?v=bG984_L6dS0&list=PL9XzOCngAkqs4m0XdULJu_78nM3Ok3Q65&index=14.

Reply:
If a service exposes an "industry standard" API of OSS software for compatibility / interoperability reasons, that may justify some exceptions to these guidelines. For now we plan to handle these on a case-by-case basis in the API reviews as we expect these will be uncommon.